scripts/build_go_apps.sh

```diff
@@ -19,7 +19,7 @@ do
     cd $directory
 
     1>&2 echo "Building test app $directory/$1.go_test_app"
-    go build -o "$1.go_test_app"
+    go build -o "$1.go_test_app" || echo "Cannot build test app $directory/$1.go_test_app"
```

Comment:
I'm not sure we want this to not fail, as it's used in CI to build our tests, and there we don't want it to pass by skipping things (unless it's explicitly skipped/ignored, like we have a couple of tests).

Why did it fail to build `go-e2e-statfs` on macos for you? Doesn't the `//go:build linux` prevent it from being built?

Reply:
When I run the script from the repository root, it tries to build both the layer integration test apps and the e2e test apps, and it fails to build `go-e2e-statfs` and quit the do loop early leaving the rest of e2e test apps not built. I just noticed that macos-test CI action works just fine because it `cd` into `mirrord/layer/tests` then run the script from there.
This is an easy tweak that can be done locally, so I removed this change.

Reply:
Since this is surprising, do you see any place to add a note about this?

Reply:
Indeed, I can add a note under the section `E2E tests`

Reply:
Added some note [here](https://github.com/metalbear-co/mirrord/pull/3236/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R97-R102)